service/history/workflow/mutable_state_impl.go

```diff
@@ -719,6 +733,10 @@ func (e *MutableStateImpl) GetCompletionEvent() (*historypb.HistoryEvent, error)
 	if err != nil {
 		return nil, err
 	}
+	lastWriteVersion, err := e.GetLastWriteVersion()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
you can merge the `GetLastWriteVersion` & above `GetCurrentBranchToken` by using line `completionEventID := e.hBuilder.NextEventID() - 1`